redfish-core/lib/processor.hpp

```diff
@@ -1160,6 +1194,110 @@ inline void handleProcessorCollectionHead(
         "</redfish/v1/JsonSchemas/ProcessorCollection/ProcessorCollection.json>; rel=describedby");
 }
 
+inline void
+    setProcessorData(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                     const dbus::utility::MapperServiceMap& serviceMap,
+                     const std::string& objectPath,
+                     std::optional<bool> locationIndicatorActive)
+{
+    if (locationIndicatorActive)
+    {
+        for (const auto& [serviceName, interfaceList] : serviceMap)
+        {
+            bool cpuInterface = false;
+            bool associationInterface = false;
+            for (const auto& interface : interfaceList)
+            {
+                if (interface == "xyz.openbmc_project.Inventory.Item.Cpu")
+                {
+                    cpuInterface = true;
+                }
+                else if (interface ==
+                         "xyz.openbmc_project.Association.Definitions")
+                {
+                    associationInterface = true;
+                }
+            }
+
+            if (cpuInterface && associationInterface)
+            {
+                setLocationIndicatorActive(asyncResp, objectPath,
+                                           *locationIndicatorActive);
+            }
+        }
+    }
+}
+
+inline void afterSetProcessorObject(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& processorId, std::optional<bool> locationIndicatorActive,
+    const boost::system::error_code& ec,
+    const dbus::utility::MapperGetSubTreeResponse& subtree)
+{
+    if (ec)
+    {
+        BMCWEB_LOG_DEBUG("DBUS response error {}", ec.value());
+        messages::internalError(asyncResp->res);
+        return;
+    }
+    for (const auto& [objectPath, serviceMap] : subtree)
+    {
+        // Ignore any objects which don't end with our desired cpu name
+        sdbusplus::message::object_path path(objectPath);
+        std::string name = path.filename();
+        if (name.empty() || name != processorId)
+        {
+            continue;
+        }
+
+        // Filter out objects that don't have the CPU-specific
+        // interfaces to make sure we can return 404 on non-CPUs (e.g.
+        // /redfish/../Processors/dimm0)
+        for (const auto& [serviceName, interfaceList] : serviceMap)
+        {
+            if (std::ranges::find_first_of(interfaceList,
+                                           processorInterfaces) !=
+                std::end(interfaceList))
```

Comment:
Can we also use this pattern to the existing code lines - L820-L841 so that it can return when matching?

L822:
```
                if (std::ranges::find_first_of(interfaceList,
                                               processorInterfaces) !=
                    std::end(interfaceList))
                {
                     handler(...);
                    return;
                }
```